nldcsc/http_apis/base_class/api_base_class.py

```diff
@@ -82,10 +82,10 @@ def _connect(
         }
 
         if data is not None:
-            if not isinstance(data, str):
-                data = json.dumps(data)
-
-            request_api_resource["data"] = data
+            if method in [self.methods.DELETE, self.methods.GET]:
+                request_api_resource["params"] = data
+            else:
+                request_api_resource["json"] = data
```

Comment:
The original code json dumped the data and added it as `data`. In this edit `json` does the conversion for you and has the added benefit of automatically setting the content-type header - either version would fail if the data is not json serializable.

But an instance test would be useful. Params accepts a dict, list of tuples or bytes. json accepts json-serializable dicts but this is hard to test based on typing. I'll  change it back to `data` to support the various types and wrap the json conversion in a try-except to catch any problematic dicts. It'll be up to the user to set the header. 

Is that okay?